### PR TITLE
LPS-185990 | Improve the UX when selecting default values for the dynamic filter

### PR DESCRIPTION
**Pull request by andreluis2022**
**Description:**
Create teste - Product QA | Functional Automation - LPS-185990 Improve the UX when selecting default values for the dynamic filter

**JIRA Ticket:**
https://liferay.atlassian.net/browse/LPS-185990

### CODE REVIEW

**Comment by andreluis2022** [comment]
ci:test:frontend-data-set

**Comment by andreluis2022** [comment]
Hey @CarlosBrichete can you take a look in these tests?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 1]
```diff
@@ -520,6 +878,88 @@ definition {
...
+				key_sortOption = "creator",
+				locator1 = "ObjectCustomViews#SORT_OPTION");
+
+			Select(
+				locator1 = "DataSet#SOURCE_OPTIONS_SELECT",
+				value1 = "Transport");
+
+			Click.clickAtNotVisible(
+				key_radioOption = "Single",
+				locator1 = "CommerceEntry#RADIO_BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:953]: Why are you using a Commerce path? Please use a generic path for radio button

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 2]
```diff
@@ -520,6 +878,88 @@ definition {
...
+		}
+
+		task ("Then there is a search in the Preselected Values dropdown") {
+			AssertElementPresent(
+				key_fieldName = "Select a default value for your filter",
+				locator1 = "ObjectCustomViews#FROM_DATE_INPUT");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:959]: Why if you are asserting the search bar you are using ObjectCustomViews#FROM_DATE_INPUT? How is this related to the search bar?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 3]
```diff
@@ -520,6 +878,88 @@ definition {
...
+		}
+
+		task ("When the user creates a new filter //Name: Filter test //Filter by: creator //Source option: Transport the selection is multiple") {
+			DataSetAdmin.goToTab(tabName = "Filters");
+
+			LexiconEntry.gotoAdd();
+
+			Click(
+				key_fieldName = "Selection",
+				locator1 = "FormViewBuilder#OBJECT_FIELD_BUTTON_DROPDOWN");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:929]: Why are you using a FormViewBuilder path? Please use a generic dropdown button path

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 4]
```diff
@@ -691,6 +1131,184 @@ definition {
...
+				key_sortOption = "creator",
+				locator1 = "ObjectCustomViews#SORT_OPTION");
+
+			Select(
+				locator1 = "DataSet#SOURCE_OPTIONS_SELECT",
+				value1 = "Transport");
+
+			Click.clickAtNotVisible(
+				key_radioOption = "Single",
+				locator1 = "CommerceEntry#RADIO_BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:1206]: Same here

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 5]
```diff
@@ -691,6 +1131,184 @@ definition {
...
+		}
+
+		task ("When the user creates a new filter //Name: Filter test //Filter by: creator //Source option: Transport the selection is multiple") {
+			DataSetAdmin.goToTab(tabName = "Filters");
+
+			LexiconEntry.gotoAdd();
+
+			Click(
+				key_fieldName = "Selection",
+				locator1 = "FormViewBuilder#OBJECT_FIELD_BUTTON_DROPDOWN");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:1182]: Same here

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 6]
```diff
@@ -691,6 +1131,184 @@ definition {
...
+			Type(
+				key_fieldName = "Select a default value for your filter",
+				locator1 = "ObjectCustomViews#FROM_DATE_INPUT",
+				value1 = "bike");
+		}
+
+		task ("Then only the existing value is displayed in the dropdown") {
+			AssertElementPresent(
+				key_sortOption = "bike",
+				locator1 = "ObjectCustomViews#SORT_OPTION");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:1219]: Same here, use a generic dropdown option path to assert this

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 7]
```diff
@@ -691,6 +1131,184 @@ definition {
...
+
+			Click.clickAtNotVisible(
+				key_radioOption = "Single",
+				locator1 = "CommerceEntry#RADIO_BUTTON");
+		}
+
+		task ("And types an existing value in Preselected Values dropdown search") {
+			Type(
+				key_fieldName = "Select a default value for your filter",
+				locator1 = "ObjectCustomViews#FROM_DATE_INPUT",
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:1212]: Same here

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 8]
```diff
@@ -330,6 +330,106 @@ definition {
...
+		}
+
+		task ("When the user creates a new filter //Name: Filter test //Filter by: creator //Source option: Transport the selection is multiple") {
+			DataSetAdmin.goToTab(tabName = "Filters");
+
+			LexiconEntry.gotoAdd();
+
+			Click(
+				key_fieldName = "Selection",
+				locator1 = "FormViewBuilder#OBJECT_FIELD_BUTTON_DROPDOWN");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:381]: Why are you using a FormViewBuilder path? Please use a generic dropdown button path

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 9]
```diff
@@ -330,6 +330,106 @@ definition {
...
+				key_sortOption = "creator",
+				locator1 = "ObjectCustomViews#SORT_OPTION");
+
+			Select(
+				locator1 = "DataSet#SOURCE_OPTIONS_SELECT",
+				value1 = "Transport");
+
+			Click.clickAtNotVisible(
+				key_radioOption = "Multiple",
+				locator1 = "CommerceEntry#RADIO_BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:405]: Why are you using a Commerce path? Please use a generic path for radio button

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 10]
```diff
@@ -330,6 +330,106 @@ definition {
...
+		}
+
+		task ("And chooses one of the values displayed in the dropdown of Preselected Values field//car") {
+			Click(
+				key_fieldName = "Select a default value for your filter",
+				locator1 = "ObjectCustomViews#FROM_DATE_INPUT");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:411]: Same here

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 11]
```diff
@@ -330,6 +330,106 @@ definition {
...
+
+			Click(
+				key_sortOption = "car",
+				locator1 = "ObjectCustomViews#SORT_OPTION");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:415]: Same here

Comment (andreluis2022) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase]: All done!

**Comment by andreluis2022** [comment]
ci:test:sf

**Comment by andreluis2022** [comment]
ci:test:frontend

**Comment by andreluis2022** [comment]
Hey @CarlosBrichete what do you think?

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/DataSet.path [hunk 12]
```diff
@@ -340,6 +355,31 @@
...
+<tr>
+	<td>PRESELECT_VALUE_INPUT</td>
+	<td>//div[contains(@class,'form-group')]//input[contains(@placeholder,'${key_fieldName}')]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>FILTER_BUTTON_DROPDOWN</td>
+	<td>//button[@class='dropdown-item' and contains(text(),'${key_fieldName}')]</td>
+	<td></td>
+</tr>
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/DataSet.path:377]: This already exists in Poshi. Please check MenuItem.path file.

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/DataSet.path [hunk 13]
```diff
@@ -340,6 +355,31 @@
...
 	<td></td>
 </tr>
+<tr>
+	<td>ITEM_CHECKBOX_CHECKED</td>
+	<td>//button[contains(@class,'dropdown-item') and contains(.,'${key_name}')]//input[@type='checkbox' and @checked]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>ANY_RADIO_BUTTON</td>
+	<td>//label[contains(.,'${key_radioOption}')]/input[@type='radio'] | //div[contains(@class,'justify')][contains(.,'${key_radioOption}')]//preceding::input[@type='radio'][1]</td>
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/DataSet.path:365]: This already exists in Poshi. Please check Button.path file.

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/DataSet.path [hunk 14]
```diff
@@ -340,6 +355,31 @@
 	<td>//div[@class="form-group"]//input[contains(@aria-label,'${key_label}')]</td>
 	<td></td>
 </tr>
+<tr>
+	<td>ITEM_CHECKBOX_CHECKED</td>
+	<td>//button[contains(@class,'dropdown-item') and contains(.,'${key_name}')]//input[@type='checkbox' and @checked]</td>
+	<td></td>
+</tr>
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/DataSet.path:362]: This already exists in Poshi. Please check Checkbox.path file.

**Review by CarlosBrichete — CHANGES_REQUESTED** [review]
Please check path again

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/DataSet.path (hunk 14 above)
Comment (andreluis2022) [reply portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/DataSet.path:362]: The checkbox is very specific to FDS filters, so it was necessary to create this path, which was only found in other components such as Object, Commerce, etc.

**Comment by andreluis2022** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/DataSet.path (hunk 14 above)
Comment (andreluis2022) [reply portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/DataSet.path]: what do you think?

**Review by CarlosBrichete — APPROVED** [review]

**Comment by CarlosBrichete** [comment]
ci:forward
